lib/p2p/Pool.ts

```diff
@@ -73,6 +74,50 @@ class Pool extends EventEmitter {
       return;
     }
 
+    if (this.listenPort) {
+      if (!handshakeData.addresses) {
+        handshakeData.addresses = [];
+      }
+
+      // Append the external IP if no address was specified by the user
+      if (handshakeData.addresses.length === 0) {
+        try {
+          // TODO: verify that this address is reachable
+          const externalip = await new Promise<string>((resolve, reject) => {
+            http.get('http://ipv4.icanhazip.com/', (res) => {
+              let body = '';
+
+              res.on('data', (chunk) => {
+                body += chunk;
+              });
+              res.on('end', () => {
+                // Removes new line at the end of the string
+                body = body.trimRight();
+                resolve(body);
+              });
+              res.on('error', (err: Error) => {
+                reject(errors.EXTERNAL_IP_UNRETRIEVABLE(err));
+              });
+
+            }).on('error', (err: Error) => {
+              reject(errors.EXTERNAL_IP_UNRETRIEVABLE(err));
+            });
+          });
+
+          this.logger.verbose(`retrieved external IP: ${externalip}`);
```

Comment:
I think debug or maybe even info would be a better logger level, I think it'd be nice when debugging and it's not like it's polluting the logs since it's at most a single line printed on startup.